libc/stdio/printf.c

```diff
@@ -219,11 +219,19 @@ int vprintf(const char *fmt, va_list args) {
       }
 
       // parse width, if it exists
-      int old_errno = errno; // save and restore errno -- printf shouldn't modify it
-      const char *width_end = fmt;
-      flags.width = strtol(fmt, (char **)&width_end, 10);
-      errno = old_errno;
-      fmt = width_end;
+      // first check for wildcard width
+      if (*fmt == '*') {
+        // read the arg
+        flags.width = va_arg(args, int);
```

Comment:
Since the argument is a signed int, we should verify that it's >= 0, and set it to 0 otherwise. We could get some unexpected behaviour in the printing code or when using `%n` otherwise.